src/main/java/space/obminyashka/items_exchange/config/SwaggerConfig.java

```diff
@@ -1,60 +1,45 @@
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.ExternalDocumentation;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.info.License;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.springdoc.core.GroupedOpenApi;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
-import org.springframework.http.HttpHeaders;
-import springfox.bean.validators.configuration.BeanValidatorPluginsConfiguration;
-import springfox.documentation.builders.ApiInfoBuilder;
-import springfox.documentation.builders.PathSelectors;
-import springfox.documentation.builders.RequestHandlerSelectors;
-import springfox.documentation.service.ApiInfo;
-import springfox.documentation.service.ApiKey;
-import springfox.documentation.service.AuthorizationScope;
-import springfox.documentation.service.SecurityReference;
-import springfox.documentation.spi.DocumentationType;
-import springfox.documentation.spi.service.contexts.SecurityContext;
-import springfox.documentation.spring.web.plugins.Docket;
-
-import java.util.List;
+import space.obminyashka.items_exchange.api.ApiKey;
 
 @Configuration
-@Import(BeanValidatorPluginsConfiguration.class)
 public class SwaggerConfig {
     @Bean
-    public Docket api() {
-        return new Docket(DocumentationType.OAS_30)
-                .useDefaultResponseMessages(false)
-                .apiInfo(apiInfo())
-                .select()
-                .apis(RequestHandlerSelectors.any())
-                .paths(PathSelectors.any())
-                .build()
-                .securitySchemes(List.of(apiKey()))
-                .securityContexts(List.of(securityContext()));
-    }
-
-    private ApiInfo apiInfo() {
-        return new ApiInfoBuilder()
-                .title("Obminyashka (Child Goods Exchange) API")
-                .description("API Definitions of the Obminyashka (Child Goods Exchange) project")
-                .version("1.0.0")
+    public GroupedOpenApi publicApi() {
+        return GroupedOpenApi.builder()
+                .group("springshop-public")
+                .pathsToMatch(ApiKey.API + "/**")
                 .build();
     }
 
-    private ApiKey apiKey() {
-        return new ApiKey("Authorization", HttpHeaders.AUTHORIZATION, "header");
-    }
-
-    private SecurityContext securityContext() {
-        return SecurityContext.builder()
-                .securityReferences(defaultAuth())
-                .build();
-    }
-
-    private List<SecurityReference> defaultAuth() {
-        return List.of(new SecurityReference(HttpHeaders.AUTHORIZATION,
-                new AuthorizationScope[]{
-                        new AuthorizationScope("global", "accessEverything")}));
+    @Bean
+    public OpenAPI springShopOpenAPI() {
```

Comment:
The name of the method has the example name and should be renamed for our needs.

Reply:
Something like that: api or obminyashkaOpenAPI?